src/pyats/contrib/creators/topology.py

```diff
@@ -0,0 +1,656 @@
+                        for net in ip_net:
+                            if ipaddress.IPv4Address(ip) in net:
+                                log.info('Ip {ip} found in ignored '
+                                         'network {net}'.format(ip = ip, 
+                                                                net = net))
+                                skip = True
+                        if skip:
+                            continue
+                    os = self.get_os(neighbor['system_description'], '')
+                    m = domain_filter.match(dest_host)
+                    if m:
+                        dest_host = m.groupdict()['hostname']
+                    # add the discovered information to both 
+                    # the connection_dict and the device_list    
+                    self.add_to_device_list(device_list, 
+                                    dest_host, 
+                                    dest_port, 
+                                    {ip_address}, 
+                                    os, 
+                                    entry)
+                    self.add_to_connection_dict(connection_dict, 
+                                            dest_host, 
+                                            dest_port, 
+                                            ip_address, 
+                                            interface,
+                                            entry)
+        return connection_dict
+
+    def add_to_device_list(self, device_list, dest_host, 
+                           dest_port, ip_address, os, discover_name):
+        '''
+            Add the information needed to create the device in the 
+            testbed later to the specified list
+        '''
+        if dest_host not in device_list:
+            device_list[dest_host] = {'ports': {dest_port}, 
+                                    'ip':ip_address, 
+                                    'os': os,
+                                    'finder': discover_name}
+        else:
+            if device_list[dest_host]['os'] is None:
+                device_list[dest_host]['os'] = os
+            device_list[dest_host]['ports'].add(dest_port)
+            device_list[dest_host]['ip'] = device_list[dest_host]['ip'].union(ip_address)
+
+    def add_to_connection_dict(self, connection_dict, 
+                               dest_host, dest_port, 
+                               ip_address,interface, dev):
+        '''
+            Adds the information about a connection to be added to the topology
+            recording what device interface combo is connected to the given
+            interface and ip address involved in the connection
+        '''
+        new_entry = {'dest_host': dest_host, 
+                    'dest_port': dest_port, 
+                    'ip_address': ip_address}
+        if interface not in connection_dict:
+            connection_dict[interface] = [new_entry]
+        else:
+            unique = True

```

Comment:
no need of this flag, 

just use for ; else

Reply:
switched to using for else loops instead of flags